train.py

```diff
@@ -681,7 +696,7 @@ def sigterm_handler(signum, frame):
     )
     logging.info(f" Grad scaler enabled: {scaler.is_enabled()} (amp mode)")
 
-    epoch_pbar = tqdm(range(args.max_epochs), position=0, leave=True)
+    epoch_pbar = tqdm(range(args.max_epochs), position=0, leave=True, dynamic_ncols=True)
```

Comment:
are you able to test this in colab and jupyter (runpod/vast)?  the pbars are touchy, I think qslug finally got them working  in vast/runpod, and don't want to break anything

Reply:
colab pbars are already kinda broke but better not to make it worse

Reply:
i'll give it a shot

Reply:
seems to work fine, except for the ANSI color codes, but that isn't anything i've touched:

<img width="1002" alt="Screen Shot 2023-03-03 at 09 50 41" src="https://user-images.githubusercontent.com/144366/222675319-d4ddc631-5669-4a8b-b866-a0ef386d568f.png">